app/src/main/java/com/example/vmedvediev/redditapp/model/Models.kt

```diff
@@ -52,4 +54,11 @@ data class Author(@field:Element(name = "name") var name: String = "",
 
 data class Post(val title: String, val author: String?, val dateUpdated: String, val postUrl: String, val thumnailUrl:String)
 
-data class Comment(val comment: String, val author: String?, val updated: String, val id: String)
\ No newline at end of file
+data class Comment(val comment: String, val author: String?, val updated: String, val id: String)
+
+data class LoginChecker(@SerializedName("json") @Expose val json: Json)
+
+data class Json(@SerializedName("data") @Expose val data: Data)
+
+data class Data(@SerializedName("modhash") @Expose val modhash: String,
```

Comment:
Annotation `@Expose` is not necessary if your field name equals to annotation key.
Check here https://stackoverflow.com/questions/34752200/gson-expose-vs-serializedname